pkg/datasource/executor.go

```diff
@@ -0,0 +1,74 @@
+package datasource
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/influxdb-flux-datasource/pkg/models"
+	influxdb2 "github.com/influxdata/influxdb-client-go"
+)
+
+func ExecuteQuery(ctx context.Context, query models.QueryModel, runner queryRunner) (dr backend.DataResponse) {
+	dr = backend.DataResponse{}
+
+	flux, err := Interpolate(query)
+	if err != nil {
+		dr.Error = err
+		return
+	}
+
+	table, err := runner.runQuery(ctx, flux)
+	if err != nil {
+		dr.Error = err
+		return
+	}
+
+	return readDataFrames(table, query.MaxDataPoints)
+}
+
+func readDataFrames(result *influxdb2.QueryTableResult, maxPoints int64) (dr backend.DataResponse) {
+	dr = backend.DataResponse{}
+
+	builder := &FrameBuilder{}
+	var points int64 = 0
+
+	for result.Next() {
+		// Observe when there is new grouping key producing new table
+		if result.TableChanged() {
+			if builder.frame != nil {
+				dr.Frames = append(dr.Frames, builder.frame)
+			}
+			err := builder.Init(result.TableMetadata())
+			if err != nil {
+				dr.Error = err
```

Comment:
It seems nitpicky, but in go, the standard seem to be to return (sr someResponse, err Error) and so I find this pattern confusing and inconsistent. I don't suppose we could still return an error, and at some later point populate it at a common place?